src/index.js

```diff
@@ -313,28 +351,28 @@ async function type(element, text, {allAtOnce = false, delay} = {}) {
       // eslint-disable-next-line no-await-in-loop
       if (delay > 0) await wait(delay)
 
-      const downEvent = fireEvent.keyDown(element, {
+      const downEvent = fireEvent.keyDown(currentElement(), {
         key,
         keyCode,
         which: keyCode,
       })
 
       if (downEvent) {
-        const pressEvent = fireEvent.keyPress(element, {
+        const pressEvent = fireEvent.keyPress(currentElement(), {
           key,
           keyCode,
           charCode: keyCode,
         })
 
         const isTextPastThreshold =
-          (actuallyTyped + key).length > (previousText + computedText).length
+          (actuallyTyped() + key).length >
+          (currentElement().maxLength || text.length)
```

Comment:
I've updated this to the way I think it should work, but I think I may have misunderstood.
The way I understand this is....
Previously we could calculate how many total characters we can type based on the value that was already in the input before we started using `previousText`. We could then use this as a check against what we've `actuallyTyped`.
Now we are using the current value (`actuallyTyped`) of the element (because the `previousValue` may no longer apply to the current element), so we're computing how many more characters we can add at any point (not total).

I had originally assumed the `isTextPastThreshold` check would be:
```const isTextPastThreshold =
          (currentValue() + key).length >
          (currentValue() + computeText()).length```
But this felt wrong, because we're checking how many more characters we can add and then adding it to the current value and that doesn't necessarily mean it's the correct length.

So, I think what I've done is correct, but let me know if I've completely missed the point 😄 
(Also, to note, where I'm saying `actuallyTyped` I've renamed it to `currentValue` in the code because this felt more reflective of the actual usage now)